src/main/flags.sh

```diff
@@ -4,19 +4,175 @@
+    FLAGS=()
+    CCLIB=("-lssl" "-lcrypto" "-lz")
+    CCOPT=("-static" "-no-pie")
 else
-  echo "( :standard )" > flags.sexp
+    case "$OS" in
+    linux)
+        if [[ -e /etc/alpine-release ]]; then
+            # The CCLIB flags statically (since we have CCOPT include -static)
+            # link in libcurl's dependencies.
+            # This can be removed when we transition away from the ocurl otel
+            # collector.
+            # old: was just '--copt -static --copy -no-pie' before we depended
+            # on libcurl
+            FLAGS=()
+            CCLIB=("-lssl" "-lcrypto" "-lz")
+            CCOPT=("-static" "-no-pie")
+        else
+            # On non-Alpine Linux distros (e.g., Ubunut), we just dynamically
+            # link for dev. Note that Alpine is used for our Docker builds.
+            FLAGS=()
+            CCLIB=()
+            CCOPT=()
+        fi
+        ;;
+    # In general, we want to statically link to make it easier to distribute
+    # binaries since they won't carry runtime dependencies on various
+    # libraries which would need to be separately installed or managed.
+    # Additionally, this can be a bit fragile depending on which paths the
+    # loader uses, and where dylibs are expected to be (and how possible
+    # differences in the CI build versus other environments).
+    #
+    # However, on MacOS, it is a bit difficult to ensure libraries are linked
+    # statically. It is not possible to link all libraries statically.
+    # Notably, libSystem cannot (officially) be statically linked against,
+    # since MacOS does not have a stable syscall ABI, but rather relies on
+    # libSystem being stable. See, e.g.,
+    # - <https://developer.apple.com/forums/thread/706419>
+    # - <https://developer.apple.com/library/archive/qa/qa1118/_index.html>
+    # - `man 1 ld`
+    #
+    # As a result, we cannot use a similar strategy to our Alpine build (which
+    # works for Linux, since it is a rare case with a stable syscall ABI, See
+    # <https://unix.stackexchange.com/questions/473137/do-other-unix-like-kernels-have-stable-syscall-abis>).
+    # Previously we relied on a workaround where we deleted the `.dylib`
+    # versions of libraries, leaving only the `.a` versions, thus forcing them
+    # to be linked statically. However, this breaks the build in the case
+    # where later build steps require programs which dynamically linked
+    # against those libraries. Notably, this occurs for PCRE2. See
+    # <https://github.com/semgrep/semgrep/pull/10064>.
+    #
+    # One might hope that we could specify the linker to use the `.a` version
+    # without deleting the `.dylib`, but the only way to do this is to disable
+    # autolinking and specify each library individually. Otherwise, on MacOS,
+    # if both are present, the linker will link dynamically regardless.
+    #
+    # If you add a dependency which should be statically linked, it should be
```

Comment:
nit: I think the "how to add a dependency" should be higher up. I wouldn't be smart enough to read this entire comment thinking there's a solution at the end, so we shouldn't bury it down here

Reply:
imo the current format here is better, since it explains why we're doing this, and the part about how to change it is closest to the part you need to change. Per your other comment, I think we could add more info at the top level, so I would rather just have a "Help I have to edit this file and don't know what to do" message there.